internal/csi-addons/networkfence/fencing_test.go

```diff
@@ -54,3 +54,43 @@ func TestGetIPRange(t *testing.T) {
 		})
 	}
 }
+
+func TestFetchIP(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		clientInfo  string
+		expectedIP  string
+		expectedErr bool
+	}{
+		{
+			clientInfo:  "client.4305 172.21.9.34:0/422650892",
+			expectedIP:  "172.21.9.34",
+			expectedErr: false,
+		},
+		{
+			clientInfo:  "",
+			expectedIP:  "",
+			expectedErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
I am not very fond of this. It would be clearer to use `tc := tt` so that `tc` is clearly an instance for the text-case. You can then use `tc` in all calls below.